pythran/pythonic/random/shuffle.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef PYTHONIC_RANDOM_SHUFFLE_HPP
+#define PYTHONIC_RANDOM_SHUFFLE_HPP
+
+#include "pythonic/include/random/shuffle.hpp"
+
+#include "pythonic/utils/proxy.hpp"
+#include "pythonic/random/random.hpp"
+
+#include <limits>
+
+namespace pythonic {
+
+    namespace random {
+
+        template <class T>
+        void shuffle(T & seq) {
+            std::shuffle(seq.begin(), seq.end(), __random_generator);
+        }
+
+        template <class T, class function>
+        void shuffle(T & seq, function && randf) {
```

Comment:
Can't you use the `std::function<T()>` type here?

(I don't know if it is possible because of functors, lambda and other advanced structures)